neqo-transport/src/connection/mod.rs

```diff
@@ -1915,6 +1921,12 @@ impl Connection {
     ) -> Res<(Vec<RecoveryToken>, bool, bool)> {
         let mut tokens = Vec::new();
         let primary = path.borrow().is_primary();
+        let mut ack_eliciting = false;
+
+        // Check if there is a Datagram to be written
+        if primary && space == PacketNumberSpace::ApplicationData && !profile.ack_only(space) {
+            ack_eliciting |= self.maybe_write_datagram(builder, &mut tokens);
+        }
```

Comment:
I for got to add the note why I did it in this way. I decided to send datagram first because ACKs can be sent even if there is a cc limit.

Anyway, I will reorder them.